hnn_qt5.py

```diff
@@ -722,19 +838,23 @@ def addtransvar (self,k,strans):
 
   def setfromdin (self,din):
     if not din: return
-    self.removeAllInputs() # turn off any previously set inputs
-    nprox, ndist = countEvokedInputs(din)
-    for i in range(nprox+ndist):
-      if i % 2 == 0:
-        if self.nprox < nprox:
-          self.addProx()
-        elif self.ndist < ndist:
-          self.addDist()
-      else:
-        if self.ndist < ndist:          
-          self.addDist()
-        elif self.nprox < nprox:
-          self.addProx()
+
+    if 'dt' in din:
```

Comment:
Optimization feature introduces the case where din just contains optimization relevant parameters. In that case, we don't want to remove all inputs, just modify existing inputs. Hence, the reason for this case.